pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -72,3 +93,62 @@ func FilterFunctionTags(input map[string]string) map[string]string {
 
 	return output
 }
+
+// RouteInferredSpan decodes the event and routes it to the correct
+// enrichment function for that event source
+func RouteInferredSpan(event string, inferredSpan InferredSpan) {
+	// Parse the event into the EventKey struct
+	eventSource, attributes := ParseEventSource(event)
+	switch eventSource {
+	case APIGATEWAY:
+		EnrichInferredSpanWithAPIGatewayRESTEvent(attributes, inferredSpan)
+	case HTTPAPI:
+		EnrichInferredSpanWithAPIGatewayHTTPEvent(attributes, inferredSpan)
+	case WEBSOCKET:
+		EnrichInferredSpanWithAPIGatewayWebsocketEvent(attributes, inferredSpan)
+	}
+}
+
+// CompleteInferredSpan finishes the inferred span and passes it
+// as an API payload to be processed by the trace agent
+func CompleteInferredSpan(
+	processTrace func(p *api.Payload),
+	endTime time.Time,
+	isError bool,
+	inferredSpan InferredSpan) {
+
+	if inferredSpan.IsAsync {
+		inferredSpan.Span.Duration = inferredSpan.CurrentInvocationStartTime.UnixNano() - inferredSpan.Span.Start
+	} else {
+		inferredSpan.Span.Duration = endTime.UnixNano() - inferredSpan.Span.Start
+	}
+	if isError {
+		inferredSpan.Span.Error = 1
+	}
+	traceChunk := &pb.TraceChunk{
+		Priority: int32(sampler.PriorityNone),
```

Comment:
@hghotra could this be a problem (regarding the sampling project ?)

Reply:
Yes, this should be coming from either:
- upstream lambda function
- the tracer (internally)